src/clr-angular/forms/select/option.spec.ts

```diff
@@ -50,29 +52,53 @@ export default function(): void {
                 expect(context.clarityElement.textContent.trim()).toMatch("Test");
             });
 
-            it("closes the menu when an item is clicked", () => {
+            it("closes the menu when an option is clicked", () => {
                 const option = context.clarityElement;
-                spyOn(context.clarityDirective, "closeMenuOnClick");
+                spyOn(context.clarityDirective, "updateSelectionAndCloseMenu");
+
+                option.click();
+
+                expect(context.clarityDirective.updateSelectionAndCloseMenu).toHaveBeenCalled();
+            });
+
+            it("adds the active class on the option when it is clicked", () => {
+                let option: HTMLElement = context.clarityElement;
+                expect(option.classList.contains("active")).toBe(false);
 
                 option.click();
+                context.detectChanges();
 
-                expect(context.clarityDirective.closeMenuOnClick).toHaveBeenCalled();
+                option = context.clarityElement;
+                expect(option.classList.contains("active")).toBe(true);
             });
         });
 
         describe("Typescript API", function() {
             beforeEach(function() {
                 context = this.create(ClrOption, TestComponent, []);
                 ifOpenService = context.getClarityProvider(IfOpenService);
+                optionSelectionService = context.getClarityProvider(OptionSelectionService);
             });
 
             it("closes the menu when an item is clicked", () => {
                 ifOpenService.open = true;
 
-                context.clarityDirective.closeMenuOnClick();
+                context.clarityDirective.updateSelectionAndCloseMenu();
 
                 expect(ifOpenService.open).toBe(false);
             });
+
+            it("calls to update the selection when an item is clicked", () => {
+                spyOn(optionSelectionService, "updateSelection");
+
+                context.clarityDirective.updateSelectionAndCloseMenu();
+
+                expect(optionSelectionService.updateSelection).toHaveBeenCalled();
+            });
+
+            it("provides a ref to the ElementRef of the option", () => {
```

Comment:
I don't see this used anywhere. Does it have a purpose or is it from trying approaches?

Reply:
We do use this in `select.ts`. We need this to clone the `option` in the input when the user makes the selection.

Reply:
Cool, page search did not show it when the file is collapsed... 😕